### PR TITLE
Temp game screens

### PR DESCRIPTION
**Pull request by holdemani**
Might be sloppy, can't remember due to the giant gap between starting and finishing. But it should all work. Has some of your branch in it. Just some

### CODE REVIEW

File: Astral Pathfinder/Game/Game.cpp [hunk 1]
```diff
@@ -61,9 +61,13 @@ void Game::init(const std::string title, SDL_Rect rect, bool fullscreen) {
     }
     
     gameState.isRunning = true;
+    gameState.mainMenu = true;
     
     // Object Initialization
```
Comment (LoganTHarvell) [inline Astral Pathfinder/Game/Game.cpp]: All of the UI stuff should prob be moved to the UIManager

File: Astral Pathfinder/Game/Game.cpp [hunk 2]
```diff
@@ -144,12 +157,23 @@ void Game::render() {
...
     SDL_RenderCopy(renderer, gameScreen, NULL, &screenRect);
     planetManager->render(&gameState);
     shipManager->render(&gameState);
   }
   
+  else if(gameState.endgame == State::allDiscovered) {
+    SDL_RenderCopy(renderer, winScreen, NULL, &screenRect);
+  }
+  
+  else if(gameState.endgame == State::fuel) {
```
Comment (LoganTHarvell) [inline Astral Pathfinder/Game/Game.cpp]: State::fuel should be renamed to noFuel if that is what it means

File: Astral Pathfinder/Game/Game.hpp [hunk 3]
```diff
@@ -85,7 +89,7 @@ class Game {
   SDL_Window *window;
   State gameState;
 
-  SDL_Texture *gameScreen;
+  SDL_Texture *mainMenu, *gameScreen, *winScreen, *loseScreen;
```
Comment (LoganTHarvell) [inline Astral Pathfinder/Game/Game.hpp]: Probably should move these all to the UIManager if possible

File: Astral Pathfinder/Game/UIManager.cpp [hunk 4]
```diff
@@ -174,3 +202,103 @@ void UIManager::checkClickedArea(SDL_Point p) {
   
   else currentWindow = none;
 }
+
+void UIManager::checkForHovering(Game::State *gs) {
```
Comment (LoganTHarvell) [inline Astral Pathfinder/Game/UIManager.cpp:236]: The hardcoded numbers should be replaced with a parameter

File: Astral Pathfinder/Game/UIManager.hpp [hunk 5]
```diff
@@ -35,22 +40,27 @@ class UIManager {
   
 private:
   // MARK: - UIManager Fields
-  // TODO: Implement main menu and endscreen
-  // MainMenu mainMenu;
   PlanetInfo selectedPlanetInfo, DockedPlanetInfo;
   ShipInfo shipInfo;
+  SDL_Texture *hoverBorder;
+  SDL_Rect borderRect = {};
```
Comment (LoganTHarvell) [inline Astral Pathfinder/Game/UIManager.hpp]: Why "= {}" ?

**Review by LoganTHarvell — DISMISSED** [review]
Needs some slight cleaning up

File: Astral Pathfinder/Game/Game.cpp (hunk 2 above)
Comment (holdemani) [reply Astral Pathfinder/Game/Game.cpp]: You're the one who named it that Q.Q but I'll change it

**Pull request merged**